packages/calcite-components/src/components/block/block.tsx

```diff
@@ -334,14 +382,42 @@ export class Block
     ) : null;
   }
 
+  renderIcon(iconPosition: string): VNode {
+    const { iconFlipRtl } = this;
+
+    if (iconPosition === undefined) {
+      return null;
+    }
+
+    const flipRtl =
+      iconFlipRtl === "both" || iconPosition === "iconStart"
+        ? iconFlipRtl === "start"
+        : iconFlipRtl === "end";
+
+    const iconValue = iconPosition === "iconStart" ? this.iconStart : this.iconEnd;
+
+    /** Icon scale is not variable as the component does not have a scale property */
+    return (
+      <calcite-icon
+        class={CSS[iconPosition]}
+        flipRtl={flipRtl}
+        icon={iconValue}
+        key={CSS[iconPosition]}
+        scale="s"
+      />
+    );
+  }
+
   render(): VNode {
     const { collapsible, el, loading, open, heading, messages } = this;
 
     const toggleLabel = open ? messages.collapse : messages.expand;
 
     const headerContent = (
       <header class={CSS.header} id={IDS.header}>
-        {this.renderIcon()}
+        {this.renderIcon(this.iconStart ? "iconStart" : undefined)}
```

Comment:
It seems odd to check for `iconStart`/`iconEnd` both outside (to determine the position) and within  `renderIcon`. I think this should be changed to one of the following:

```tsx
{this.renderIcon("start")} // knows the position, so it can check for the start icon and render if possible
```

or 

```tsx
{this.renderIcon(this.iconStart, "start")} // renders whatever icon and position is passed
```

Leaning more towards the first one as it takes care of matching the icon for us.